### PR TITLE
Update Travis CI to use Xenial for most jobs

### PR DESCRIPTION
**Pull request by felixdivo**
- Moves Travis from Trusty to Xenial for most jobs (except OSX jobs & testing socketcan, see below)
- testing socketcan remains as a test on Python 3.6 only on Trusty
- Makes building the docs run in a separate job
- Fixes various smaller problems with skipping socketcan tests outside of Travis CI
- Fixes: `Warning: 'classifiers' should be a list, got type 'tuple'` on Python 3.8a ("nightly")

### CODE REVIEW

**Comment by felixdivo** [comment]
This is done. Do you agree with the changes? Python 3.8 (dev & nightly) fails for some reason but that's not a problem for today.

**Comment by felixdivo** [comment]
TODO: [Apparently](https://docs.codecov.io/docs/merging-reports) the coverage should be uploaded on all builds an codecov.io merges the reports, which results in a high coverage being reported. I will change it back to how it was.

File: .travis.yml [hunk 1]
```diff
@@ -25,39 +26,54 @@ os:
...
 sudo: required
 
 matrix:
-  # see "os: ..." above
   include:
+    # building the docs
+    - python: "3.7"
+      env: BUILD_ONLY_DOCS=TRUE
+    # testing socketcan on Trusty & Python 3.6, since it is not available on Xenial
+    - os: linux
```
Comment (hardbyte) [inline .travis.yml]: It might just be this one that needs sudo?

File: .travis.yml [hunk 2]
```diff
@@ -25,39 +26,54 @@ os:
...
+    if [[ "$BUILD_ONLY_DOCS" ]]; then
+      # Build the docs with Sphinx
+      #   -a Write all files
+      #   -n nitpicky
+      python -m sphinx -an doc build
+    else
+      # Run the tests
+      python setup.py test && # (preserve the error code semantics)
+      # Upload the coverage to codecov.io
+      if [[ "$TRAVIS_PYTHON_VERSION" == "3.7" ]]; then codecov; fi
```
Comment (hardbyte) [inline .travis.yml]: As you commented, this condition needs to be removed - code cov should also be updated when the tests fail. Won't the `&&` short circuit?

File: .travis.yml (hunk 1 above)
Comment (felixdivo) [reply .travis.yml]: Ah yes, that could work.

File: .travis.yml (hunk 2 above)
Comment (felixdivo) [reply .travis.yml]: Jup, this now works.

**Pull request merged**